internal/engine/wazevo/backend/isa/arm64/lower_mem_test.go

```diff
@@ -847,3 +848,74 @@ func Test_extLoadSizeSign(t *testing.T) {
 		require.Equal(t, tc.signed, signed)
 	}
 }
+
+func Test_lowerLoadSplatFromAddressMode(t *testing.T) {
+	positiveTests := make(map[addressModeKind]bool)
+	nextVReg := regalloc.VReg(100).SetRegType(regalloc.RegTypeInt)
+
+	for _, tc := range []struct {
+		amode       addressMode
+		expected    string
+		expectPanic bool
+	}{
+		{
+			amode: addressMode{kind: addressModeKindRegReg, rn: v0VReg, rm: v1VReg},
+			expected: `
+add x100?, d0, d1
+ld1r {x10.4s}, [x100?]
+`,
+		},
+		{
+			amode: addressMode{kind: addressModeKindRegUnsignedImm12, rn: v0VReg, imm: 42},
+			expected: `
+add x100?, d0, #0x2a
+ld1r {x10.4s}, [x100?]
+`,
+		},
+		{
+			amode: addressMode{kind: addressModeKindRegSignedImm9, rn: v0VReg, imm: 42},
+			expected: `
+add x100?, d0, #0x2a
+ld1r {x10.4s}, [x100?]
+`,
+		},
+		{
+			amode: addressMode{kind: addressModeKindRegSignedImm9, rn: v0VReg, imm: 42},
+			expected: `
+add x100?, d0, #0x2a
+ld1r {x10.4s}, [x100?]
+`,
+		},
+	} {
+		tc := tc
+		t.Run("address mode "+strconv.Itoa(int(tc.amode.kind)), func(t *testing.T) {
+			ctx, _, m := newSetupWithMockContext()
+			ctx.vRegCounter = int(nextVReg.ID()) - 1
+			positiveTests[tc.amode.kind] = true
+
+			m.lowerLoadSplatFromAddressMode(operandNR(x10VReg), tc.amode, ssa.VecLaneI32x4)
+			require.Equal(t, tc.expected, "\n"+formatEmittedInstructionsInCurrentBlock(m)+"\n")
+		})
+	}
+
+	// Must panic for all other addressModeKinds.
+	for k := 0; k <= int(addressModeKindResultStackSpace); k++ {
+		amk := addressModeKind(k)
+		if positiveTests[amk] {
+			continue
+		}
+
+		ctx, _, m := newSetupWithMockContext()
+		ctx.vRegCounter = int(nextVReg.ID()) - 1
+
+		t.Run("address mode "+strconv.Itoa(k), func(t *testing.T) {
+			defer func() {
+				errMsg := recover().(string)
```

Comment:
This is not the way to test panic (see other places in the codebase)

Reply:
ooh, neat `require.CapturePanic()`